### DISCUSSION TITLE
Is entt useful for non-game software?

### DISCUSSION THREAD

**Discussion by Idealist-dreamer** [skypjack/entt#1200] [Q&A]
I'm using entt to build EDA software.Maybe this is strange, but my original idea was to use entt to build more scalable and performant software.

1. Scalability
EDA software contains many modules, and different modules need to use smart Pointers to frequently reference each other. After using entt, different modules only need to bind its specific components on the entity.

2. High performance
Use entt to speed up special system update processes such as rendering.

The above is the starting point for me to use entt. Could you give me some professional advice?

3. I didn't learn entt very much, so I didn't know much about the deeper signaling and event mechanics of ENTT.I tried to use components as event tags, bind them to entities, and then retrieve and process them by other systems.Is it efficient to remove the event tag to indicate that processing is complete?
4. I also have a problem, my software needs to be undone and restored, so I often use registry.create and registry.destry. During this process, the entity returned by create has increased by ten times. I am new to ecs, I don't understand why it has leapfrocked. Since the user's undo recovery can be up to hundreds of times, I am concerned that it is beyond the range of uint32_t. Is this a problem?

Thank you for reading and replying.

**Comment by skypjack** [comment]
> Is it efficient to remove the event tag to indicate that processing is complete?

Yes, it is. It depends on the implementation actually. EnTT is designed to make it the default because I use this approach **a lot**. 🙂 

> Since the user's undo recovery can be up to hundreds of times, I am concerned that it is beyond the range of uint32_t. Is this a problem?

What changes is the version actually. Try to look at `entt::to_entity(identifier)` and `entt::to_version(identifier)`.
This isn't a problem anyway because the version wraps to 0 after a while. You can find more details in the documentation. 👍
